test/tests/functional/pbs_qrun.py

```diff
@@ -92,3 +94,44 @@ def test_invalid_host_val(self):
         msg = "Server is not up"
         self.assertTrue(self.server.isUp(), msg)
         self.logger.info("As expected server is up and running")
+
+    def test_qrun_hangs(self):
+        """
+        This test submit 500 jobs with differnt equivalence class,
+        turn of scheduling and qrun job to
+        verify whether qrun hangs.
+        """
+        node = self.mom.shortname
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduling': 'False'})
+        self.server.manager(MGR_CMD_SET, NODE,
+                            {'resources_available.ncpus': 1}, id=node)
+        for walltime in range(1, 501):
+            j = Job(TEST_USER)
+            a = {'Resource_List.walltime': walltime}
+            j.set_attributes(a)
+            if walltime == 500:
+                jid = self.server.submit(j)
+            else:
+                self.server.submit(j)
+        self.logger.info("Submitted 500 jobs with different walltime")
+
+        now = int(time.time())
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduling': 'True'})
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduling': 'False'})
+        pid = os.fork()
+        if pid > 0:
```

Comment:
wait, why are you making the parent process execute the runjob command? So the child process is calling os.kill() on the parent? Why are you doing that?

Reply:
Addressed this in my recent commit.